app/src/main/java/org/covidwatch/android/ui/home/HomeFragment.kt

```diff
@@ -30,7 +43,17 @@ class HomeFragment : BaseFragment<FragmentHomeBinding>() {
         homeViewModel.navigateToOnboardingEvent.observe(viewLifecycleOwner, EventObserver {
             findNavController().navigate(R.id.splashFragment)
         })
+
+        if (RandomEnObjects.retrieved == true) {
```

Comment:
My assumption is that the problems that you mentioned in the previous PR due to the incorrect usage of the `PreferenceStorage`. You created a new instance of it the repository while the `HomeFragment` uses different one. That might've messed up things but I'm not certain. Try my recommendation above and just return `HomeFragment` to state as it used to be before this PR.

If it doesn't work there should be another way of doing this. And perhaps there is a bug in the `SharedPreferenceStorage `

Reply:
I don't understand this

Reply:
I fixed the problem I had previously.